### PR TITLE
Fix the link failure in the English custom-resource-definition-v1.md

### PR DESCRIPTION
**Pull request by lishaokai1995**
Signed-off-by: sky li (李少凯)-浪潮数据 <lishaokai1995@126.com>
English custom-resource-definition-v1.md has invalid link, I fixed it

### CODE REVIEW

**Comment by mengjiao-liu** [comment]
Related PR: https://github.com/kubernetes/website/pull/36916
See https://github.com/kubernetes/website/pull/36916#issuecomment-1251952928

**Comment by kbhawkey** [comment]
Hello @lishaokai1995 .
This page is generated using the Kubernetes source code. The link error may be present in the code.

Page preview:
https://deploy-preview-36919--kubernetes-io-main-staging.netlify.app/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/#CustomResourceDefinitionSpec

**Comment by mengjiao-liu** [comment]
Yes, I didn't notice yesterday that this file is in the reference directory, the link needs to be modified within the k8s source code, and then the website repo needs new reference docs.

I submitted PR [#1122](https://github.com/kubernetes/kubernetes/pull/112615) in the Kubernetes repo to update this link.

File: content/en/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1.md [hunk 1]
```diff
@@ -276,7 +276,7 @@ CustomResourceDefinitionSpec describes how a user wants their resource to appear
 
 - **preserveUnknownFields** (boolean)
 
-  preserveUnknownFields indicates that object fields which are not specified in the OpenAPI schema should be preserved when persisting to storage. apiVersion, kind, metadata and known fields inside metadata are always preserved. This field is deprecated in favor of setting `x-preserve-unknown-fields` to true in `spec.versions[*].schema.openAPIV3Schema`. See https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#pruning-versus-preserving-unknown-fields for details.
+  preserveUnknownFields indicates that object fields which are not specified in the OpenAPI schema should be preserved when persisting to storage. apiVersion, kind, metadata and known fields inside metadata are always preserved. This field is deprecated in favor of setting `x-preserve-unknown-fields` to true in `spec.versions[*].schema.openAPIV3Schema`. See https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning for details.
```
Comment (chrisnegus) [inline content/en/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1.md:279]:
```suggestion
  preserveUnknownFields indicates that object fields which are not specified in the OpenAPI schema should be preserved when persisting to storage. apiVersion, kind, metadata and known fields inside metadata are always preserved. This field is deprecated in favor of setting `x-preserve-unknown-fields` to true in `spec.versions[*].schema.openAPIV3Schema`. See [Field pruning](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning) for details.
```

**Review by chrisnegus — COMMENTED** [review]
Just a suggestion to identify the heading name and link directly within the docs.

**Comment by kbhawkey** [comment]
> Yes, I didn't notice yesterday that this file is in the reference directory, the link needs to be modified within the k8s source code, and then the website repo needs new reference docs.
> 
> I submitted PR [kubernetes/kubernetes#112615](https://github.com/kubernetes/kubernetes/pull/112615) in the Kubernetes repo to update this link.

Thanks for following up with another pull request!

**Comment by kbhawkey** [comment]
Good suggestion about linking directly to docs. Though, I think the link is found in the source code.

/approve

**Pull request merged**